### ISSUE TITLE
Extending Feelix to work with the Moteus brushless controller

### ISSUE THREAD

**Issue by Maxw3llGM** [ankevanoosterhout/Feelix2.0#2]
Just opening a thread here to discuss the process.

I have been getting to understand how Feelix works. Does the microcontroller that is loaded up with the Feelix Firmware communicate directly to the motor using [simpleFOC](https://simplefoc.com/)? 

![image](https://github.com/ankevanoosterhout/Feelix2.0/assets/87669716/e2173f9f-dc0b-40c9-b955-820aa2808176)

From what I have understood so far, I will attempt to create a wrapper so that the Feelix Software can communicate with the Moteus controller by transmitting the information like torque, angle and position over.

**Comment by ankevanoosterhout** [comment]
Yes, the microcontroller directly controls the motor. For a project I am implementing also possibilities to have one microcontroller as master that distributes data to a bunch of microcontrollers that each control a motor but I need to make a few adjustments for that still.

The figure is right, the microcontroller directly communicates with the driver and encoder, it reads the sensorposition and then sends updates to the driver and sends sensor data back to Feelix (less frequently). Feelix sends all the data of the effects at the start. You could send the data you need to the ESP_32 and then forward the (modified) data to the moteus controller. I believe you don't need the simpleFOC library the moteus controllers seems to handle that bit for you? 

If you want I can send some driver boards, I have two that I just tested. In case you run into issues with the compatibility with the moteus controller. It is a board with a L6234 driver (https://docs.feelix.xyz/hardware-support/connect-microcontroller) which is a bit more powerful then my new boards and you can connect it with the ESP. In case you want to use the driver board you need to check if your esp 32 is compatible with simplefoc https://docs.simplefoc.com/esp_mcu.

**Comment by Maxw3llGM** [comment]
It would be great if you could send two driver boards! I'm pretty sure SimpleFoc is compatible with tinypico, if not we have a tiny pico I can also use. Moteus does handle what SimpleFOC does, the board takes as input torque, velocity and/or position values. So to extend feelix to TorqueTuner, something similar to the principal/agent setup you are using will be done, most likely the will instead not be converted to voltage values as that will be handled by the Moteus board.

I'm still looking into how to make the conversion.
